target/ppc/translate/fp-impl.c.inc

```diff
@@ -1473,6 +1473,52 @@ static void gen_stfqx(DisasContext *ctx)
     tcg_temp_free_i64(t1);
 }
 
+static bool resolve_PLF_D(DisasContext *ctx, arg_PLF_D *a)
+{
+    if (a->r) {
+        if (unlikely(a->ra != 0)) {
+            gen_invalid(ctx);
+            return false;
+        }
+        a->si += ctx->cia;
+    }
+    return true;
+}
+
+static bool ea_calc(DisasContext *ctx, int ra, int r, TCGv displ, TCGv *ea)
+{
+
+    if (ra) {
+        tcg_gen_add_tl(*ea, cpu_gpr[ra], displ);
+    } else {
+        tcg_gen_mov_tl(*ea, displ);
+    }
+    if (NARROW_MODE(ctx)) {
+        tcg_gen_ext32u_tl(*ea, *ea);
+    }
+
+    return true;
+}
+
+static bool trans_PLFS(DisasContext *ctx, arg_PLFS *a)
+{
+    TCGv ea;
+    TCGv_i64 t0;   
+    if (!resolve_PLF_D(ctx, a)) {
```

Comment:
I think you need to check `ctx->fpu_enabled` like in `gen_lfs`. It's probably a good idea to create a macro for this check, as they requested for `ctx->altivec_enabled`.